src/guards/routes.ts

```diff
@@ -0,0 +1,49 @@
+import { combineName } from '@/services/combineName'
+import { RouterRoute, isRouterRoute } from '@/services/createRouterRoute'
+import { RegisteredRouteMap, RegisteredRoutesKey } from '@/types/register'
+import { ResolvedRoute } from '@/types/resolved'
+
+export type IsRouteOptions = {
+  exact?: boolean,
+}
+
+export function isRoute(route: unknown): route is RouterRoute
+export function isRoute<TRouteKey extends RegisteredRoutesKey>(route: unknown, routeKey: TRouteKey, options: IsRouteOptions): route is RouterRoute<ResolvedRoute<RegisteredRouteMap[TRouteKey]>>
+export function isRoute(route: unknown, routeKey?: string, { exact }: IsRouteOptions = {}): boolean {
+  if (!isRouterRoute(route)) {
+    return false
+  }
+
+  if (routeKey === undefined) {
+    return true
+  }
+
+  const keys = getRouteKeys(route)
+
+  if (exact) {
+    const actualRouteKey = keys.at(-1)
+
+    return routeKey === actualRouteKey
+  }
+
+  return keys.includes(routeKey)
+}
+
+function getRouteKeys(route: RouterRoute): string[] {
```

Comment:
I moved this from `useRoute.ts` and modified it to accept the route itself rather than the names.